### ISSUE TITLE
Gradle-Plugin: Tasks with TypeResolution are not using the same compiler options as configured in the project

### ISSUE THREAD

**Issue by mervyn-mccreight** [detekt/detekt#3639]
Hi folks :-)

When using tasks like e.g. `detektMain` I figured out that it displays a lot of compiler warnings.

E.g. I have opted-in kotlin.time by passing a compiler-flag like the following in my `build.gradle.kts`
```kotlin
tasks.withType<KotlinCompile> {
    kotlinOptions.freeCompilerArgs += "-Xuse-experimental=kotlin.time.ExperimentalTime"
}
```
so I do not have to opt-in it all over the source-code.

`detektMain` does not seem to pick this up and is constantly complaining about it with messages like this:

```
<path/to/source/File.kt>: error: this declaration is experimental and its usage must be marked with '@kotlin.time.ExperimentalTime' or '@OptIn(kotlin.time.ExperimentalTime::class)
```

which does not look very nice. Is the validation even working for these files?
The task itself "passes" if no source-file violates any rules, but in my perception it is misleading that the terminal is full of red text containing the term "error" if everything is fine.

Is it intended that detekt is not using the same compiler flag as the project uses to check against rules using type-resolution?
Or am I missing some configuration so that detekt does this?

_Originally posted by @mervyn-mccreight in https://github.com/detekt/detekt/discussions/3629_

**Comment by chao2zhang** [comment]
Hello @mervyn-mccreight, these compiler warnings are issued by kotlin compiler, and detekt internally also uses the kotlin compiler for a lightweight compilation pass. This indicate that detekt may not pick up the compiler flags specified in your projects.

These are purely command-line compiler warnings and errors and should not affect the result of the detekt Gradle task. In the long term, we would need to move Detekt to compiler plugin like https://github.com/detekt/detekt-compiler-plugin but in the short term, we can also expose options to allow addition compiler flags in our detekt gradle task.

**Comment by cortinico** [comment]
This will be fixed by #4106
